src/index.ts

```diff
@@ -67,65 +63,84 @@ export function generateWords(nr: number, words: string[] = []): string {
     nr = Math.abs(nr);
   }
 
+  let remainder = 0;
+  let quotient: number;
+  let integer: number;
+  let decimals: number;
+
   switch (true) {
     case (nr < 20):
-      remainder = 0;
-      word = LESS_THAN_TWENTY[Math.trunc(nr)];
-      word += parseDecimals(nr);
+      [,, integer, decimals] = parseNumber(nr);
+      words.push(LESS_THAN_TWENTY[integer]);
+      if (decimals > 0) {
+        words.push(parseDecimals(decimals));
+      }
       break;
     case (nr < ONE_HUNDRED):
-      remainder = Math.trunc(nr % TEN);
-      word = TENTHS_LESS_THAN_HUNDRED[Math.floor(nr / TEN)];
+      [quotient, remainder] = parseNumber(nr, TEN);
```

Comment:
adding to my previous comment where you had [,, integer, decimals] now your returned array has only  [quotient, remainder], fancy but hard to keep track of what is the expected return, in the first instance you give it a number and only the last 2 values will be defined, now you give it 2 numbers and your array will have 4 values defined but you are interested only in the first 2, consider using  {} instead of [] at least it makes it more readable and predictable.